fms/utils/tokenizers.py

```diff
@@ -67,6 +70,10 @@ def convert_ids_to_tokens(self, ids: torch.LongTensor):
         return [chr(i) for i in ids]
 
     def convert_tokens_to_ids(self, tokens: Union[str, list[str]]):
+        if isinstance(tokens, str) and len(tokens) == 1:
```

Comment:
hmm... so for the case where `len(tokens)!=1`, we're treating a string as a list of tokens (below) for this tokenizer.

I believe the interface for the other tokenizers is that a string parameter is treated as a single token, and only a list is treated as a list of tokens. We should aim to make the behavior consistent across tokenizers, otherwise users could run into issues when switching between them.

Reply:
the potential issue is that if tokens is a multi-character string, `token_id = ord(tokens)` would raise `ord() expected a character, but string of length 2 found`. Assuming we're disallowing multi-token strings, we might want to explicitly raise a more meaningful error message.

Reply:
Just added this check. Should I also add the check in the case where we are given a list of strings (line 81)?

Reply:
if the user passed a list of strings and one is multi-char, they're probably using the API correctly but have a data issue. In that case it might be best to just return 0 for that token.

we should grep through the code for any other uses of char_tokenizer too just in case there's any code without unit tests that might be using the API incorrectly

Reply:
Okay, in line 81 I returned 0 for those tokens and added a test to check that in the test_tokenizers file. Also checked for other uses of char_tokenizer and think they all work correctly.